grin/src/Transformations/Optimising/NonSharedElimination.hs

```diff
@@ -12,12 +12,17 @@ import qualified Data.Set as Set
 
 import Grin.Grin
 import Grin.TypeEnv
+import Transformations.Names (ExpChanges(..))
 import AbstractInterpretation.Sharing.Result
 
 
 
-nonSharedElimination :: SharingResult -> TypeEnv -> Exp -> Exp
-nonSharedElimination SharingResult{..} te = cata skipUpdate where
+nonSharedElimination :: SharingResult -> TypeEnv -> Exp -> (Exp, ExpChanges)
+nonSharedElimination SharingResult{..} te exp = (exp', change) where
+
+  exp' = cata skipUpdate exp
+
+  change = if exp' /= exp then DeletedHeapOperation else NoChange
```

Comment:
Change should be computed differently. But this one is for later.

Reply:
Alright.